visidata/loaders/json.py

```diff
@@ -126,13 +168,49 @@ def write_jsonl(vs, fp):
             for row in vs.iterrows():
                 rowdict = _rowdict(vcols, row)
                 fp.write(jsonenc.encode(rowdict) + '\n')
+        if len(vs) == 0:
+            vd.warning(
+                "Output file is empty - cannot save headers without data for jsonl.\n"
+                "Use `options.json_save_arrays = True` to save as JSONL arrays format "
+                "rather than JSONL dict format to preserve the headers."
+            )
+
+
+def get_jsonla_rows(sheet, cols):
+    for row in Progress(sheet.rows):
+        yield [col.getTypedValue(row) for col in cols]
+
+
+@Sheet.api
+def write_jsonla(vs, fp):
+        vcols = vs.visibleCols
+        jsonenc = _vjsonEncoder()
+        with Progress(gerund='saving'):
+            header = [col.name for col in vcols]
+            fp.write(jsonenc.encode(header) + '\n')
+            rows = get_jsonla_rows(vs, vcols)
+            for row in rows:
+                fp.write(jsonenc.encode(row) + '\n')
 
 
 @VisiData.api
 def save_jsonl(vd, p, *vsheets):
     with p.open_text(mode='w', encoding=vsheets[0].options.encoding) as fp:
         for vs in vsheets:
-            vs.write_jsonl(fp)
+            if vs.options.json_save_arrays:
+                vd.status("Saving JSONL arrays")
+                vs.write_jsonla(fp)
+            else:
+                vd.status("Saving JSONL dicts")
+                vs.write_jsonl(fp)
+
+
+@VisiData.api
+def save_jsonla(vd, p, *vsheets):
+    with p.open_text(mode='w', encoding=vsheets[0].options.encoding) as fp:
+        for vs in vsheets:
+            vd.status("Saving JSONL arrays")
```

Comment:
Let's remove these extra `vd.status()` in this function

Reply:
Done.